libpromises/evalfunction.c

```diff
@@ -619,44 +676,39 @@ static bool CallHostsSeenCallback(const char *hostkey, const char *address,
                                   ARG_UNUSED bool incoming, const KeyHostSeen *quality,
                                   void *ctx)
 {
-    Item **addresses = ctx;
-
     if (HostKeyAddressUnknown(hostkey))
     {
         return true;
     }
```

Comment:
Do we still want to omit these hosts in all the new cases?

Reply:
You mean that we could ignore the fact that the address is uknown if we are asking for the shas? If so, we can easily move this check to GetHostsFromLastseenDB if the user asks for addersses.
```c
case ADDRESS:
{
    if (HostKeyAddressUnkown(hd->hostkey))
    {
        continue;
    }
    ...
}
```
Doesnt sound too unreasonable, however im not sure what the implications of an uknown address are...

Reply:
I think it makes sense to return the hostkey (sha) if its known and user asked for it.

Reply:
Should there be a "guard" in the [switch](https://github.com/cfengine/core/blob/0bfc9529b4cefa2a4c58d5e14229f9e8db2a901d/libpromises/evalfunction.c#L579-L595) if the case is NAME or ADDRESS to deal with any unknown addresses?

Reply:
As a user, I don't care about the implementation ;) But if I ask for hosts by hostkey, I'd expect to get all hosts that match the criteria and they hostkey is known.